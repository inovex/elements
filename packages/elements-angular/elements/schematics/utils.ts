import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
