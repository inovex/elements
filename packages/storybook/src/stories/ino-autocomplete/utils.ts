import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
