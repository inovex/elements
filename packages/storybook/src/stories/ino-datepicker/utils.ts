import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
