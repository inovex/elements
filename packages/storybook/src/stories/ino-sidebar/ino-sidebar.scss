@use '../../../../elements/src/components/base/theme';

$small-shadow: 0 4px 8px 0 rgba(61, 64, 245, 0.3),
0 12px 28px 0 rgba(61, 214, 245, 0.14);

body {
  margin: 0;
}

.sidebar-iframe {
  width: 100%;
  height: 400px;
  border: 1px theme.color(light, base) solid;
}

.sidebar-demo {
  display: block;
  height: 100%;

  ino-icon {
    --icon-color: #fff;
  }

  .header {
    position: absolute;
    width: 100%;
    display: flex;
    height: 40px;
    background-color: theme.$brand;
    justify-content: flex-start;
    flex-direction: row;
    padding: 0 10px 0 10px;

    &--section {
      justify-content: flex-start;
      display: inline-flex;
      align-items: center;
      color: #fff;
      font-size: 22px;

      p {
        padding: 0 0 0 20px;
        color: #fff;
      }
    }
  }

  .content {
    padding-top: 40px;
  }
}

.sidebar-header {
  display: flex;
  height: 60px;
  flex-direction: row;
  justify-content: flex-end;

  ino-icon {
    --icon-color: #0d10f3;
  }
}

ino-sidebar[align-right] {
  .sidebar-header {
    justify-content: flex-start;
  }
}

