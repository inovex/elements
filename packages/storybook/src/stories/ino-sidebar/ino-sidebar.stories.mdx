import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-sidebar/readme.md";

<Meta title="Structure/ino-sidebar/Documentation" />

<Description markdown={readme} />
