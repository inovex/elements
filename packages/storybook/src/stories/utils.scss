/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
