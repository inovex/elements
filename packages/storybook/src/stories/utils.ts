import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
