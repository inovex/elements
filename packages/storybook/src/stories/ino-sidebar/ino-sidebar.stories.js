import { number } from '@storybook/addon-knobs';

import withStencilReadme from '_local-storybookcore/with-stencil-readme';

import componentReadme from '_local-elements/src/components/ino-sidebar/readme.md';
import './ino-sidebar.scss';

function subscribeToComponentEvents() {
  // == event block

  const handleEvent = function (e) {
    const el = e.target;
    if (el.tagName.toLowerCase() !== 'ino-icon') {
      return;
    }
    const sidebar = document.querySelector('ino-sidebar');
    sidebar.inoOpen = !sidebar.inoOpen;
  };

  document.addEventListener('clickEl', handleEvent);
  // == event block

  // unsubscribe function will be called by Storybook
  return () => {
    document.removeEventListener('clickEl', handleEvent);
  };
}

export default {
  title: 'Structure/<ino-sidebar>',

  decorators: [
    (story) => {
      subscribeToComponentEvents();
      return story();
    },
  ],
};

export const DefaultUsage = () => {
  return /*html*/ `
    `;
};

DefaultUsage.decorators = [withStencilReadme(componentReadme)];

export const SidebarLeft = () => {
  return `
    <div class="sidebar-demo">
      ${header}
      <ino-sidebar>
        ${sidebarContent}
      </ino-sidebar>
      ${mainContent}
    </div>
    `;
};

export const SidebarRight = () => {
  return `
    <div class="sidebar-demo">
      ${header}
      <ino-sidebar ino-align-right>
        ${sidebarContent}
      </ino-sidebar>
      ${mainContent}
    </div>
    `;
};

export const SidebarDifferentWidth = () => {
  return `
    <div class="sidebar-demo">
      ${header}
      <ino-sidebar style="--ino-sidebar-width:${number(
        '--ino-sidebar-width',
        500,
      )}px;">
        ${sidebarContent}
      </ino-sidebar>
      ${mainContent}
    </div>
    `;
};

const header = `
<div class="header">
    <div class="header--section">
        <ino-icon ino-clickable ino-icon="menu"></ino-icon>
        <p>Sidebar</p>
    </div>
</div>
`;

const sidebarContent = `

<div class="sidebar-header" slot="header">
    <ino-icon ino-clickable ino-icon="close"/>
</div>
<div class="sidebar-content" slot="content">
  <ino-list>
    <ino-list-item ino-text="List item"></ino-list-item>
    <ino-list-item ino-text="List item"></ino-list-item>
    <ino-list-item ino-text="List item"></ino-list-item>
    <ino-list-item ino-text="List item"></ino-list-item>
    <ino-list-item ino-text="List item"></ino-list-item>
    <ino-list-item ino-text="List item"></ino-list-item>
  </ino-list>
</div>`;

const mainContent = `
<div class="content">
    <p>Cras magna diam, dictum a pretium non, elementum at nibh. Etiam
    laoreet suscipit dui et feugiat. Vivamus id consectetur dolor. Morbi eget nunc vel felis gravida scelerisque.
    Proin libero erat, suscipit pretium venenatis vitae, fringilla sed enim. Maecenas elit est, finibus in tellus
    id, interdum hendrerit ligula. Nam vitae sagittis enim. Donec pulvinar vel quam a posuere. Maecenas egestas
    ex erat, quis convallis turpis blandit a.</p>
    <p>Maecenas sodales mauris vitae feugiat egestas. Morbi
    porta laoreet odio vel semper. Quisque rutrum varius nulla. Morbi sagittis metus quis nibh congue, eu
    vehicula dui congue. Pellentesque ullamcorper eros in nulla interdum auctor. Pellentesque congue, felis vitae
    tempor porta, ipsum felis viverra dui, vel porttitor elit nibh sed purus. Nam eu diam magna. Sed nec nunc
    lorem. Suspendisse vulputate odio lacus, eget dignissim augue pharetra eget. Ut quis augue eu elit venenatis
    faucibus. Nam ut commodo dui. .</p>
    <p>Praesent eget augue sed nibh aliquam lacinia ac in lectus. Vivamus sapien urna,
    pretium in turpis at, dapibus mattis orci.Maecenas sodales mauris vitae feugiat egestas. Morbi
    porta laoreet odio vel semper. Quisque rutrum varius nulla. Morbi sagittis metus quis nibh congue, eu
    vehicula dui congue. Pellentesque ullamcorper eros in nulla interdum auctor. Pellentesque congue, felis vitae
    tempor porta, ipsum felis viverra dui, vel porttitor elit nibh sed purus. Nam eu diam magna. Sed nec nunc
    lorem. Suspendisse vulputate odio lacus, eget dignissim augue pharetra eget. Ut quis augue eu elit venenatis
    faucibus. Nam ut commodo dui. Praesent eget augue sed nibh aliquam lacinia ac in lectus. Vivamus sapien urna,
    pretium in turpis at, dapibus mattis orci</p>
    <p>Proin sem nisi, tincidunt eget ipsum non, vestibulum feugiat tellus. Sed nunc enim, sodales condimentum
    hendrerit eleifend, aliquam ac eros. Maecenas ut molestie ligula. Aenean semper nunc felis, at cursus neque congue
    in. Sed cursus mauris nunc, et scelerisque leo ornare nec. In nec dui eu sem sodales sodales ac at enim. Integer
    sit amet libero vitae magna facilisis malesuada sagittis nec dolor. Aenean eleifend dapibus ante, id accumsan
    turpis luctus varius. Etiam accumsan tortor ex, sit amet suscipit turpis vehicula et. Ut ultrices ex nisi, eget
    tristique nulla tempor eget. Etiam sed vehicula velit. Nullam lacus libero, convallis in imperdiet sit amet,
    hendrerit sed nunc.</p>
  </div>
`;
