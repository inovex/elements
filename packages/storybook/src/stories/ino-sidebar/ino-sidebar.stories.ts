import { Components } from '@inovex.de/elements';
import { useEffect } from '@storybook/client-api';
import { Story } from '@storybook/web-components';
import { html } from 'lit-html';
import { defaultDecorator } from '../utils';
import './ino-sidebar.scss';

const eventHandler = (e) => {
  const el: HTMLElement = e.target;
  const elTagName = el.tagName.toLowerCase();

  console.log(el);
  if (elTagName !== 'ino-icon' && elTagName !== 'ino-button') {
    return;
  }

  const sidebar: HTMLInoSidebarElement = el
    .closest('.sidebar-demo')
    .querySelector('ino-sidebar');

  sidebar.open = !sidebar.open;
};

export default {
  title: 'Structure/ino-sidebar',
  component: 'ino-sidebar',
  decorators: [
    (story) => defaultDecorator(story, 'story-sidebar'),
    (story) => {
      useEffect(() => {
        document.addEventListener('click', eventHandler);
        return () => {
          document.removeEventListener('click', eventHandler);
        };
      });
      return story();
    },
  ],
};

export const Playground: Story<Components.InoSidebar> = (args) => html`
  <div class="sidebar-demo">
    <ino-button>Toggle Sidebar</ino-button>
    <ino-sidebar
      id="customizable-sidebar"
      align-right="${args.alignRight}"
      name="${args.name}"
      open="${args.open}"
    >
      ${sidebarContent}
    </ino-sidebar>
  </div>
`;

Playground.args = {
  alignRight: false,
  name: '',
  open: false,
};

export const SidebarLeft = () => html`
  <div class="sidebar-demo">
    ${header}
    <ino-sidebar open> ${sidebarContent}</ino-sidebar>
    ${mainContent}
  </div>
`;

export const SidebarRight = () => html`
  <div class="sidebar-demo">
    ${header}
    <ino-sidebar open align-right> ${sidebarContent}</ino-sidebar>
    ${mainContent}
  </div>
`;

export const SidebarDifferentWidth = (args) => html`
  <div class="sidebar-demo">
    ${header}
    <ino-sidebar open style="--ino-sidebar-width:${args.inoSidebarWidth}">
      ${sidebarContent}
    </ino-sidebar>
    ${mainContent}
  </div>
`;
SidebarDifferentWidth.args = {
  inoSidebarWidth: '500px',
};

const header = html`
  <div class="header">
    <div class="header--section">
      <ino-icon clickable icon="menu"></ino-icon>
      <p>Sidebar</p>
    </div>
  </div>
`;

const sidebarContent = html`
  <div class="sidebar-header" slot="header">
    <ino-icon clickable icon="close" />
  </div>
  <div class="sidebar-content" slot="content">
    <ino-list>
      <ino-list-item text="List item"></ino-list-item>
      <ino-list-item text="List item"></ino-list-item>
      <ino-list-item text="List item"></ino-list-item>
      <ino-list-item text="List item"></ino-list-item>
      <ino-list-item text="List item"></ino-list-item>
      <ino-list-item text="List item"></ino-list-item>
    </ino-list>
  </div>
`;

const mainContent = html`
  <div class="content">
    <p>
      Cras magna diam, dictum a pretium non, elementum at nibh. Etiam laoreet
      suscipit dui et feugiat. Vivamus id consectetur dolor. Morbi eget nunc vel
      felis gravida scelerisque. Proin libero erat, suscipit pretium venenatis
      vitae, fringilla sed enim. Maecenas elit est, finibus in tellus id,
      interdum hendrerit ligula. Nam vitae sagittis enim. Donec pulvinar vel
      quam a posuere. Maecenas egestas ex erat, quis convallis turpis blandit a.
    </p>
    <p>
      Maecenas sodales mauris vitae feugiat egestas. Morbi porta laoreet odio
      vel semper. Quisque rutrum varius nulla. Morbi sagittis metus quis nibh
      congue, eu vehicula dui congue. Pellentesque ullamcorper eros in nulla
      interdum auctor. Pellentesque congue, felis vitae tempor porta, ipsum
      felis viverra dui, vel porttitor elit nibh sed purus. Nam eu diam magna.
      Sed nec nunc lorem. Suspendisse vulputate odio lacus, eget dignissim augue
      pharetra eget. Ut quis augue eu elit venenatis faucibus. Nam ut commodo
      dui. .
    </p>
    <p>
      Praesent eget augue sed nibh aliquam lacinia ac in lectus. Vivamus sapien
      urna, pretium in turpis at, dapibus mattis orci.Maecenas sodales mauris
      vitae feugiat egestas. Morbi porta laoreet odio vel semper. Quisque rutrum
      varius nulla. Morbi sagittis metus quis nibh congue, eu vehicula dui
      congue. Pellentesque ullamcorper eros in nulla interdum auctor.
      Pellentesque congue, felis vitae tempor porta, ipsum felis viverra dui,
      vel porttitor elit nibh sed purus. Nam eu diam magna. Sed nec nunc lorem.
      Suspendisse vulputate odio lacus, eget dignissim augue pharetra eget. Ut
      quis augue eu elit venenatis faucibus. Nam ut commodo dui. Praesent eget
      augue sed nibh aliquam lacinia ac in lectus. Vivamus sapien urna, pretium
      in turpis at, dapibus mattis orci
    </p>
    <p>
      Proin sem nisi, tincidunt eget ipsum non, vestibulum feugiat tellus. Sed
      nunc enim, sodales condimentum hendrerit eleifend, aliquam ac eros.
      Maecenas ut molestie ligula. Aenean semper nunc felis, at cursus neque
      congue in. Sed cursus mauris nunc, et scelerisque leo ornare nec. In nec
      dui eu sem sodales sodales ac at enim. Integer sit amet libero vitae magna
      facilisis malesuada sagittis nec dolor. Aenean eleifend dapibus ante, id
      accumsan turpis luctus varius. Etiam accumsan tortor ex, sit amet suscipit
      turpis vehicula et. Ut ultrices ex nisi, eget tristique nulla tempor eget.
      Etiam sed vehicula velit. Nullam lacus libero, convallis in imperdiet sit
      amet, hendrerit sed nunc.
    </p>
  </div>
`;
