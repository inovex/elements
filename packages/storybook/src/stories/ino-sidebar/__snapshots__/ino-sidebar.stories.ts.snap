// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structure/ino-sidebar Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-sidebar">
    <div class="sidebar-demo">
      <ino-button>
        Toggle Sidebar
      </ino-button>
      <ino-sidebar id="customizable-sidebar"
                   align-right="false"
                   name
                   open="false"
      >
        <div class="sidebar-header"
             slot="header"
        >
          <ino-icon clickable
                    icon="close"
          >
          </ino-icon>
        </div>
        <div class="sidebar-content"
             slot="content"
        >
          <ino-list>
            <ino-list-item text="List item">
            </ino-list-item>
            <ino-list-item text="List item">
            </ino-list-item>
            <ino-list-item text="List item">
            </ino-list-item>
            <ino-list-item text="List item">
            </ino-list-item>
            <ino-list-item text="List item">
            </ino-list-item>
            <ino-list-item text="List item">
            </ino-list-item>
          </ino-list>
        </div>
      </ino-sidebar>
    </div>
  </div>
</div>
`;
